### PR TITLE
M3-6041: Add Awareness of DX tools (API-CLI ) - Create Linode flow

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝
Introduces API CLI Awareness Feature to cloud manager

**What does this PR do?**
Enables users copy  and execute dynamic cURL and CLI commands in create Linode flow.

## Preview 📷

https://user-images.githubusercontent.com/119517080/214948403-614f3fe7-55c2-4ddc-b5e9-1e91e7c69e52.mov

## How to test 🧪
- Click Create Linode 
- Click Create Using Command Line button
- Should be able to inline error messages
- Fill all the required fields to create Linode
- Now, click on Create Using Command Line button
- Should open API CLI Modal
- Validate the copy and Modal as per the design
- Copy the command (cURL / CLI )
- Open terminal and paste
- Update the command with token
- Execute and observe the Linode's page should appear created Linode

**How do I run relevant unit or e2e tests?**

- yarn test
- yarn cy:run

### CODE REVIEW

File: packages/manager/src/utilities/generate-cURL.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  it('should return a curl command that has an HTTP method set', () => {
+    expect(generatedCommand).toMatch(/-X (POST)/);
+  });
+
+  it('should return a curl command that has a data option set', () => {
+    expect(generatedCommand).toMatch('-d');
+  });
+
+  it.skip('should return a curl command that has a data option set to the argument passed to it', () => {
```
Comment (cpathipa) [inline packages/manager/src/utilities/generate-cURL.test.ts:44]: @CodyFinn-Akamai Are we skipping it intentionally?

File: packages/manager/src/utilities/generate-cli.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+const generatedCommand = generateCLICommand(linodeData);
+
+describe('generateCLICommand', () => {
+  it('should return a linode-cli linodes create command', () => {
+    expect(
+      generatedCommand.startsWith('linode-cli linodes create')
+    ).toBeTruthy();
+  });
+
+  it.skip('should return a linode-cli command with the data provided formatted as arguments', () => {
```
Comment (cpathipa) [inline packages/manager/src/utilities/generate-cli.test.ts:38]: @CodyFinn-Akamai  Are we skipping it intentionally?

File: packages/manager/src/utilities/generate-cli.test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    expect(generatedCommand).not.toMatch('undefined');
+  });
+
+  it('should parse array fields as multiple command args with the same key but diffeerent values', () => {
+    expect(generatedCommand).toMatch('--authorized_users Linny');
+    expect(generatedCommand).toMatch('--authorized_users Gritty');
+  });
+
+  describe('parsing of string arguments', () => {
+    it.skip('should escape strings that contain single quotes, double quotes, and forward slashes', () => {
```
Comment (cpathipa) [inline packages/manager/src/utilities/generate-cli.test.ts:52]: @CodyFinn-Akamai Same here as well

File: packages/manager/src/utilities/generate-cURL.test.ts (hunk 1 above)
Comment (CodyFinn-Akamai) [reply packages/manager/src/utilities/generate-cURL.test.ts:44]: Yes they aren't set up to handle the heavy use of escaping and right now they provide limited use

**Review by dwiley-akamai — APPROVED** [review]

**Comment by bnussman-akamai** [comment]
One thing i'm noticing is that the generated CLI command does not like the tag `okay!\test'okay`. cURL looks good.

**Review by carrillo-erik — APPROVED** [review]

**Comment by carrillo-erik** [comment]
> One thing i'm noticing is that the generated CLI command does not like the tag `okay!\test'okay`. cURL looks good.

I verified the changes you merged and both tag and root password look for both cURL and CLI command.

**Comment by cpathipa** [comment]
Tested with some complex strings API is working as expected and able to login into VM as well.

**Pull request merged**